File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    ans = 0
    same, diff = 0, 0
    check = ''
    for c in s:
        if same == diff:
            ans += 1
            check = c

        if c == check:
            same += 1
        else:
            diff += 1
    return ans

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int ans = 0;
        int same = 0, diff = 0;
        String check = "";

        for (String c : s.split("")){
            if (same == diff) {
                ans++;
                check = c;
            }

            if (c.equals(check)) same++;
            else diff++;
        }
        return ans;
    }
}